File: packages/shared/types/insights.ts
export interface Narrative {
    title: string;
    story: string;
    audience_segment: string;
}

export interface Opportunity {
    title: string;
    description: string;
    trend_momentum: number;
    days_to_event: number;
}

export interface Scenario {
    title: string;
    description: string;
    feasibility_status: string;
    prep_checklist: string[];
}

export interface InsightsBundle {
    narratives: Narrative[];
    opportunities: Opportunity[];
    scenarios: Scenario[];
}

export interface ArtisanProfile {
    name: string;
    craft: string;
    lineage: string;
    voice_note_url: string;
}

File: apps/api/schemas/insights.py
from pydantic import BaseModel
from typing import List

class Narrative(BaseModel):
    title: str
    story: str
    audience_segment: str

class Opportunity(BaseModel):
    name: str
    channel_type: str
    why_now: str
    who_to_contact: str
    entry_script: str
    feasibility: str
    packaging: str
    timing_window: str
    evidence: List[str]
    sources: List[str]
    novelty_score: int
    momentum_score: int
    feasibility_score: int
    moq_guidance: str

class Scenario(BaseModel):
    title: str
    description: str
    feasibility_status: str
    prep_checklist: List[str]

class InsightsBundle(BaseModel):
    narratives: List[Narrative]
    opportunities: List[Opportunity]
    scenarios: List[Scenario]

class MentorAdviseRequest(BaseModel):
    prompt: str
    chat_id: int # New field
